testgrid/cmd/summarizer/example/client.go

```diff
@@ -0,0 +1,167 @@
+	serverAddr         = flag.String("server_addr", "127.0.0.1:10000", "The server address in the format of host:port")
+	serverHostOverride = flag.String("server_host_override", "x.test.youtube.com", "The server name use to verify the hostname returned by TLS handshake")
+)
+
+func collectTestResult() (*responsepb.Response, error) {
+	// Mock data. When building an application client, replace with
+	// real test result aggregated into the Response struct
+	mockTestRows := []*responsepb.Row{
+		{
+			Name:       "fake-test-1",
+			Target:     "fake-test-1-target",
+			Messages:   []string{"FooError", "", "BarException"},
+			LinkedBugs: []string{"123", "456"},
+			Alert: &responsepb.TestAlert{
+				FailCount:   1,
+				FailTime:    2,
+				PassTime:    3,
+				Link:        "fake-link-1",
+				LinkText:    "fake-link-1-text",
+				UrlText:     "url",
+				FailBuildId: "100",
+				Message:     "BarException",
+				TestId:      "fake-id-1",
+			},
+		},
+		{
+			Name:       "fake-test-2",
+			Target:     "fake-test-2-target",
+			Messages:   []string{},
+			LinkedBugs: []string{},
+			Alert: &responsepb.TestAlert{
+				FailCount:   4,
+				FailTime:    5,
+				PassTime:    6,
+				Link:        "fake-link-2",
+				LinkText:    "fake-link-2-text",
+				UrlText:     "url",
+				FailBuildId: "200",
+				Message:     "",
+				TestId:      "",
+			},
+		},
+	}
+	mockAlertOptions := configpb.DashboardTabAlertOptions{
+		AlertMailToAddresses: "mock-to@",
+		NumFailuresToAlert:   2,
+	}
+	mockDashboardTabConfig := configpb.DashboardTab{
+		Name:           "tab",
+		TestGroupName:  "test-group",
+		AlertOptions:   &mockAlertOptions,
+		CodeSearchPath: "code-path/",
+	}
+	mockQueryData := responsepb.Response{
+		Alerts:        "fake-alert",
+		LatestGreen:   "50",
+		OverallStatus: summarypb.DashboardTabSummary_FAIL,
+		Summary:       "fake-summary",
+		Tests:         mockTestRows,
+		Timestamps:    []int64{3, 2, 1},
+		OpenTestTemplate: &configpb.LinkTemplate{
+			Url: "https://url.com/target",
+			Options: []*configpb.LinkOptionsTemplate{
+				{
+					Key:   "id",
+					Value: "<test-id>",
+				},
+				{
+					Key:   "target",
+					Value: "<test-name>",
+				},
+			},
+		},
+		DashboardTab:    &mockDashboardTabConfig,
+		DashboardName:   "dashboard",
+		UpdateTimestamp: time.Now().Unix(),
+	}
+
+	return &mockQueryData, nil
+}
+
+func getSummary(c grpcpb.SummarizerServiceClient) {
+	logrus.Println("Get the translated summary object")
+	query, err := collectTestResult()
+	if err != nil {
+		logrus.Fatalf("failed to fetch test results: %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	request := &grpcpb.SummarizerRequest{
+		Response: query,
+	}
+	response, err := c.GetSummary(ctx, request)
+	if err != nil {
+		logrus.Fatalf("%v.GetSummary() failed, %v", c, err)
+	}
+	logrus.Println(response)
+}
+
+func main() {
+	logrus.Println("Starting the Client")
+	flag.Parse()
+	var opts []grpc.DialOption
```

Comment:
This seems to do similar things as testgrid's resultstore client: https://github.com/kubernetes/test-infra/blob/master/testgrid/resultstore/client.go#L34

Reply:
It kind does. This is a reference implementation for the Summarizer client. Summarizer is a standalone service, and this client is here for the reviewer to try and understand the component, and for others to use as a reference when building the real client to interact with the summarizer server.